### PR TITLE
feat: Implements hann_window for torch frontend

### LINKED ISSUE #21912: Hann_Window

### PR DESCRIPTION
**Pull request by Enigmafiend**
Close #21912

### CODE REVIEW

**Comment by Enigmafiend** [comment]
ivy-gardener

File: ivy_tests/test_ivy/test_frontends/test_torch/test_spectral_ops.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+@handle_frontend_test(
+    fn_tree="torch.hann_window",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=["int32", "int64"],
+        max_num_dims=0,
+        min_value=5,
+        max_value=10,
+    ),
+    periodic=st.booleans(),
+    # dtype=helpers.get_dtypes("float", full=False),
```
Comment (YushaArif99) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_spectral_ops.py:19]: again, this seems to be the same error as reviewed in a different PR (https://github.com/unifyai/ivy/pull/21731#discussion_r1318162968)

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Left a comment 🙂

**Comment by Enigmafiend** [comment]
ivy-gardener

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @Enigmafiend 

Thanks for making the requested changes. The tests also seem to be passsing now. Just one final thing, could you kindly fix the linting errors [here](https://github.com/unifyai/ivy/actions/runs/6155860015/job/16703443012?pr=21913#step:5:144) before I go ahead with the merge? Thanks! 🙂

**Comment by YushaArif99** [comment]
ivy-gardener

**Comment by Enigmafiend** [comment]
ivy-gardener

**Comment by Enigmafiend** [comment]
The lint error that the test fails for seems to not even exist, it conforms to the lint rules as needed.

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @Enigmafiend 

In order to fix the linting errors, you'll have to run `pre-commit` on the modified files and then push the changes. 

so the command you need to run will be 
```python

pre-commit run --files ivy/functional/frontends/torch/spectral_ops.py ivy_tests/test_ivy/test_frontends/test_torch/test_spectral_ops.py

```

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Pull request closed without merging**